packages/api/lib/writeRecords/write-granules.js

```diff
@@ -518,6 +518,89 @@ const writeGranuleRecordAndPublishSns = async ({
   });
 };
 
+/**
+ * Update granule record status in DynamoDB, PostgreSQL, Elasticsearch.
+ * Publish SNS event for updated granule.
+ *
+ * @param {Object}  params
+ * @param {Object}  params.apiGranuleRecord      - API Granule object to write to
+ *                                                 the database
+ * @param {Object}  params.postgresGranule       - PostgreSQL granule record to write
+ *                                                 to database
+ * @param {Object}  params.fieldUpdates                 - DynamoDB fields to update
+ * @param {Object}  params.dynamoFieldsToDelete         - DynamoDB fields to delete
+ * @param {Object}  params.granuleModel          - Instance of DynamoDB granule model
+ * @param {Object}  params.granulePgModel        - @cumulus/db compatible granule module instance
+ * @param {Knex}    params.knex                  - Knex object
+ * @param {string}  params.snsEventType          - SNS Event Type, defaults to 'Update'
+ * @param {Object}  params.esClient              - Elasticsearch client
+ * returns {Promise}
+ */
+const _updateGranule = async ({
+  apiGranuleRecord,
+  postgresGranule,
+  fieldUpdates,
+  dynamoFieldsToDelete,
+  granuleModel,
+  granulePgModel,
+  knex,
+  snsEventType = 'Update',
+  esClient,
+}) => {
+  const esGranule = {
+    ...apiGranuleRecord,
+  };
+  const granuleId = apiGranuleRecord.granuleId;
+  delete esGranule.execution;
+
+  let updatedPgGranule;
+  const status = 'queued';
+  await createRejectableTransaction(knex, async (trx) => {
+    [updatedPgGranule] = await granulePgModel.update(
+      trx,
+      { cumulus_id: postgresGranule.cumulus_id },
+      { status },
```

Comment:
I don't think we want this function to have to assume a specific `status`, but we also can't apply `fieldUpdates` here directly because field names in Dynamo and PG may not be the same. So perhaps we also need a `pgFieldUpdates` field. And we should rename `fieldUpdates` to `apiFieldUpdates`

Reply:
[`21ea079` (#2593)](https://github.com/nasa/cumulus/pull/2593/commits/21ea0795e9f8eafaf839a99b7341acb262176ad5)